rust/arrow/src/ffi.rs

```diff
@@ -193,6 +216,42 @@ fn to_datatype(format: &str) -> Result<DataType> {
         "ttm" => DataType::Time32(TimeUnit::Millisecond),
         "ttu" => DataType::Time64(TimeUnit::Microsecond),
         "ttn" => DataType::Time64(TimeUnit::Nanosecond),
+
+        // Note: The datatype null will only be created when called from ArrowArray::buffer_len
+        // at that point the child data is not yet known, but it is also not required to determine
+        // the buffer length of the list arrays.
+        "+l" => {
+            let nullable = schema.flags == 2;
```

Comment:
Should be `schema.flags & 2 != 0`. Also I'm surprised you're sprinkling magic numbers in the code instead of defining a constant.